### PR TITLE
[BugFix] Fix Be crash in ASAN mode if load a unapplied rowset from disk

### PR DESCRIPTION
**Pull request by sevev**
## What type of PR is this：
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes https://github.com/StarRocks/StarRocksTest/issues/2343

## Problem Summary(Required) ：

The following code in `TabletUpdates`
```
    for (auto& [rsid, rowset] : _rowsets) {
        auto stats = std::make_unique<RowsetStats>();
        stats->num_segments = rowset->num_segments();
        stats->num_rows = rowset->num_rows();
        stats->byte_size = rowset->data_disk_size();
        stats->num_dels = 0;
        for (int i = 0; i < rowset->num_segments(); i++) {
            auto itr = del_vector_cardinality_by_rssid.find(rsid + i);
            if (itr != del_vector_cardinality_by_rssid.end() && itr->second != -1) {
                stats->num_dels += itr->second;
            } else {
                std::string msg = strings::Substitute("delvec not found for rowset $0 segment $1", rsid, i);
                LOG(ERROR) << msg;
                DCHECK(false); // BE will crash in this line if we find a unapplied rowset 
            }
        }
        DCHECK_LE(stats->num_dels, stats->num_rows) << " tabletid:" << _tablet.tablet_id() << " rowset:" << rsid;
        _calc_compaction_score(stats.get());
        std::lock_guard lg(_rowset_stats_lock);
        _rowset_stats.emplace(rsid, std::move(stats));
    }
```
The `del_vector_cardinality_by_rssid` keep the all applied rowset and `_rowsets` keep the all commited rowset.  So if a rowset already commit but not apply, we can not find the rowset in `del_vector_cardinality_by_rssid` because this rowset does not have delete vector and this will cause BE crash in ASAN mode.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: be/src/storage/tablet_updates.cpp [hunk 1]
```diff
@@ -267,14 +267,16 @@ Status TabletUpdates::_load_from_pb(const TabletUpdatesPB& tablet_updates_pb) {
...
-                DCHECK(false);
+        if (unapplied_rowsets.find(rsid) == unapplied_rowsets.end()) {
+            for (int i = 0; i < rowset->num_segments(); i++) {
+                auto itr = del_vector_cardinality_by_rssid.find(rsid + i);
+                if (itr != del_vector_cardinality_by_rssid.end() && itr->second != -1) {
+                    stats->num_dels += itr->second;
+                } else {
+                    std::string msg = strings::Substitute("delvec not found for rowset $0 segment $1", rsid, i);
+                    LOG(ERROR) << msg;
+                    DCHECK(false);
```
Comment (imay) [inline be/src/storage/tablet_updates.cpp:278]: Is it appropriate to return an error status here?

File: be/src/storage/tablet_updates.cpp [hunk 2]
```diff
@@ -267,14 +267,16 @@ Status TabletUpdates::_load_from_pb(const TabletUpdatesPB& tablet_updates_pb) {
...
         stats->num_dels = 0;
-        for (int i = 0; i < rowset->num_segments(); i++) {
-            auto itr = del_vector_cardinality_by_rssid.find(rsid + i);
-            if (itr != del_vector_cardinality_by_rssid.end() && itr->second != -1) {
-                stats->num_dels += itr->second;
-            } else {
-                std::string msg = strings::Substitute("delvec not found for rowset $0 segment $1", rsid, i);
-                LOG(ERROR) << msg;
-                DCHECK(false);
+        if (unapplied_rowsets.find(rsid) == unapplied_rowsets.end()) {
```
Comment (imay) [inline be/src/storage/tablet_updates.cpp:270]: To give some comment on why this check is needed?

Comment (sevev) [reply be/src/storage/tablet_updates.cpp:271]: > To give some comment on why this check is needed?

done.

File: be/src/storage/tablet_updates.cpp (hunk 1 above)
Comment (sevev) [reply be/src/storage/tablet_updates.cpp]: > Is it appropriate to return an error status here?

done

**Review by imay — APPROVED** [review]

**Review by decster — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by yongbingwang** [comment]
branch-2.5
https://github.com/StarRocks/starrocks/pull/22425

**Comment by sevev** [comment]
@mergify backport branch-2.3

**Pull request merged**